pkg/user/signer_test.go

```diff
@@ -69,12 +71,44 @@ func (s *SignerTestSuite) TestSubmitTx() {
 	require.EqualValues(t, 0, resp.Code)
 }
 
-func (s *SignerTestSuite) ConfirmTxTimeout() {
-	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+// TestConfirmTx verifies that the ConfirmTx method returns the expected results.
+func (s *SignerTestSuite) TestConfirmTx() {
+	t := s.T()
+
+	// 1. ConfirmTx returns an error when the context time out
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+	_, err := s.signer.ConfirmTx(ctx, "E32BD15CAF57AF15D17B0D63CF4E63A9835DD1CEBB059C335C79586BC3013728")
+	assert.Error(t, err)
+	assert.Contains(t, err.Error(), context.DeadlineExceeded.Error())
+
+	// 2. ConfirmTx returns an error when tx is not found
+	ctx, cancel = context.WithTimeout(context.Background(), 5*time.Second)
 	defer cancel()
-	_, err := s.signer.ConfirmTx(ctx, string("E32BD15CAF57AF15D17B0D63CF4E63A9835DD1CEBB059C335C79586BC3013728"))
-	require.Error(s.T(), err)
-	require.Equal(s.T(), err, context.DeadlineExceeded)
+	_, err = s.signer.ConfirmTx(ctx, "not found tx")
+	assert.Error(t, err)
+
+	// 3. ConfirmTx has no error if tx is found immediately
+	fee := user.SetFee(1e6)
+	gas := user.SetGasLimit(1e6)
+	msg := bank.NewMsgSend(s.signer.Address(), testnode.RandomAddress().(sdk.AccAddress), sdk.NewCoins(sdk.NewInt64Coin(app.BondDenom, 10)))
+	resp, err := s.submitTxWithoutConfirm([]sdk.Msg{msg}, fee, gas)
+	assert.NoError(t, err)
+	assert.NotNil(t, resp)
+	resp, err = s.signer.ConfirmTx(s.ctx.GoContext(), resp.TxHash)
+	assert.NoError(t, err)
+	assert.Equal(t, resp.Code, uint32(0))
```

Comment:
[nit] the order of arguments to assert.Equal is important for debugging. It should be expected then actual so that if this test fails, the logs help debug.

```suggestion
	assert.Equal(t, abci.CodeTypeOK, resp.Code)
```

Reply:
Thanks